src/hooks/useActionRemoveRow.js

```diff
@@ -0,0 +1,24 @@
+// import InitialGridConfig from "../config/theme/initial_grid";
+// import { GridCell } from "../util/generate_grid_nodes";
+
+const useActionRemoveRow = (gridNodes, setGridNodes) => {
+  const handleSelect = () => {
+
+    // if length of gridNodes.length is not zero --> gridNodes pop
+    // if it is zero (or no more gridNodes left), this also make sure it doesn't give any out of bound error
+    // for the user it makes no difference. can press button as many times as uder desires
+    if (!gridNodes.length == 0){ 
+      gridNodes.pop();
```

Comment:
Not a good practice to modify the state (actually array and objects as well) directly unless it's explicitly documented. You should instead create another array variable which would be a clone of the original `gridNodes` variable similar to how you're setting the state in line 14. The difference is that you make the copy of the array first and then modify that copied array.

You probably won't face any issue related to that in this application cause it's pretty small but the main reason for that would be race condition. If another function was also working with gridNodes at the time that this was being executed, there would be corrupted data.

Reply:
@shikdernyc  please check my new commit where I made a copy of gridNodes instead
and I just checked warnings and this is the warning I have that I believe is the one stopping my code form passing the checks here.
the warning is referring to the following  line of my code
 if (!gridNodesCopy.length == 0)
what is your advice here? because using === signs is not working here. From what I understand === works when both entities that are being compared are strictly the same type. so doesn't the .length function in an array return an int as same as what we are comparing to an 0?
lmk what your advice is in fixing the following warning. thanks
<img width="822" alt="Screen Shot 2021-03-18 at 3 13 07 PM" src="https://user-images.githubusercontent.com/43054296/111683957-8357a680-87fc-11eb-9ddb-1378b026c3e1.png">

Reply:
`===` should work. You can use `===` for any comparison. With triple equal, it makes sure the type AND the value of both side of the equation are equal. For example (` 0 === 0`) is true.

With double equal `==`, it'll attempt to convert both the value to the same type before comparison. For example, `0=="0"` is true. The string 0 is converted to number before comprision.

In the case of `<Array>.length`, the type of this is an integer, and the type of `0` is an integer, so you can use triple equal for comparison. You can, of course, use the double equal sign as well and it will work but it's best practice to use triple-equal in this case. With React, it'll complain if you don't and Netlify won't build if React is complaining.

Reply:
thanks, @shikdernyc 
yes I do understand that and it should work even with '==='. 
as per the warning, I replaced '==' with '===' and while the warnings are no longer 
the remove Row functionality is no longer working after replacing it with ===
while before it was working
while no errors show up, why is it that replacing '==' with '===' isn't working and not allowing for the comparison to happen which in hand doesn't allow gridNodes to pop and remove?
why is '==' working and not '===' in this line of code that compares?

the following is with the change of '==='
https://user-images.githubusercontent.com/43054296/111693271-cb7bc680-8806-11eb-8a44-662ba00bfb88.mov

this is the original code before with '==' which is working

https://user-images.githubusercontent.com/43054296/111693963-a50a5b00-8807-11eb-9ec7-24056775d13f.mov

Reply:
Because you have `!` before `gridNodes`, turning it into a boolean type. So effective you're comparing `boolean === number`, which is always false. It should be `(gridNodes.length !== 0)`

Reply:
oh thanks, @shikdernyc 
can't believe I overlooked something so simple!
anyways just updated please approve pr! thanks!